crates/lsp/src/lsp.rs

```diff
@@ -164,6 +164,33 @@ struct Error {
     message: String,
 }
 
+#[derive(Debug)]
+pub enum ServerStatus {}
```

Comment:
Let's add some rustdoc, pointing at https://github.com/rust-lang/rust-analyzer/blob/master/docs/dev/lsp-extensions.md#server-status since we picked it as a baseline for this.